Three times PTOs allows PTO to expire (#4084)
quicwg/base-drafts: draft-ietf-quic-transport.md

   validation_timeout = max(3*PTO, 6*kInitialRtt)
~~~

This timeout allows for multiple PTOs to expire prior to failing path

> validation, ensuring one packet loss does not cause path validation failure.

Reviewer: ```suggestion
validation, so that loss of a single PATH_CHALLENGE frame does not cause path
validation failure.
```

To avoid excessively small idle timeout periods, endpoints MUST increase the
idle timeout period to be at least three times the current Probe Timeout (PTO).

> This allows for multiple PTOs to expire prior to idle timeout, allowing for at
> least one packet to be lost prior to idle timeout.

Reviewer: Allow for recovery, not loss.

```suggestion
This allows for multiple PTOs to expire prior to idle timeout, allowing for the endpoint
to recover any packets lost during this time.
```

Reviewer 2: Perhaps "ensuring a single packet loss event cannot cause an idle timeout"? We certainly can't recover from *any* degree of packet loss.

Author: I like your suggestion, taken.

To avoid excessively small idle timeout periods, endpoints MUST increase the
idle timeout period to be at least three times the current Probe Timeout (PTO).

> This allows for multiple PTOs to expire prior to idle timeout, ensuring a single
> packet loss event cannot cause an idle timeout.

Reviewer: That's not right, since loss events don't cause idle timeout.
```suggestion
This allows for multiple PTOs to expire during the idle timeout period, allowing
a sender to potentially recover lost packets.
```

Author: For idle timeout, it doesn't matter if the packets are recovered.

Reviewer: I don't understand this sentence -- how can the idle timeout occur as a result of a single packet loss?

Reviewer 2: A too-short idle timer might expire before the loss of a single packet that would otherwise have reset the timer could lead to a probe being sent.

Reviewer: @Ralith : I'm having trouble parsing your sentence, but a PTO does not reset the idle timer.

Reviewer 2: Receiving a probe does.
